pulp_ansible/app/tasks/collections.py

```diff
@@ -608,13 +663,56 @@ async def _find_all_collections(self):
 
         await asyncio.gather(*tasks)
 
+    async def _can_be_optimized(self):
+        """Check last synced metadata time."""
+        root, api_version = await self._get_root_api(self.remote.url)
+        if api_version == 3:
+            downloader = self.remote.get_downloader(
+                url=root, silence_errors_for_response_status_codes={404}
+            )
+            try:
+                metadata = parse_metadata(await downloader.run())
+            except FileNotFoundError:
+                return False
+
+            sources = set()
+            if self.collection_info:
+                sources = {r.source for r in self.collection_info if r.source}
+            sources.add(self.remote.url)
+            has_multiple_sources = len(sources) > 1
+
+            published = parse_datetime(metadata.get("published", ""))
+
+            if published == self.repository.last_synced_metadata_time and not has_multiple_sources:
+                msg = _(
+                    "no-op: {repo} did not change since last sync - {published}".format(
```

Comment:
I'm suggesting to make this a report that the no-op was checked, not that it occurred. So that would have it be reporting in all cases unless optimize=False at which point I expect Pulp not to evaluate the no-op case at all.

Reply:
I don't see the value of having no-op check in progress reports

Reply:
If you want to take it out I'm ok with that. My thought was that if we don't have any progress report and the no-op runs the user has no info about why. Also it tells the user what was performed differently when they call `optimize=False`.

What we're doing right now though doesn't seem right (to me). Usually progress reports report on what occurred in the task code, but this is more like a message to the user about why the task reached its completed state.

Reply:
@daviddavis wdyt?

Reply:
> So that would have it be reporting in all cases unless optimize=False at which point I expect Pulp not to evaluate the no-op case at all.

This makes sense to me.

> I'm suggesting to make this a report that the no-op was checked, not that it occurred.

Would this also contain the result of the no-op check? If not, is there some other way a user could determine if a sync task was no-op?

Reply:
With what I imagined (which the test would actually be asserting on) the only way the user would know is they would see the progress report that the no-op was checked and that there are no other progress reports, e.g. metadata downloading, etc. That's not really explicit though.

Reply:
It's not explicit but it's probably still obvious enough for users. I don't have a strong opinion here other than thinking we should show something to users even if it's just showing that there was a no-op check performed.